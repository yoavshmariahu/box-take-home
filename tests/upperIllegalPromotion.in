k a1
K e5
p a2
G d5

[G S B R]
[p r b s]

move a2 a3
move d5 c5 promote
