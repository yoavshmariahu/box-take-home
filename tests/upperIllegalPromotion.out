UPPER player action: move d5 c5 promote
5 |__|__|__| G| K|
4 |__|__|__|__|__|
3 | p|__|__|__|__|
2 |__|__|__|__|__|
1 | k|__|__|__|__|
    a  b  c  d  e

Captures UPPER: G S B R
Captures lower: p r b s

lower player wins.  Illegal move.
